File: infra/protocol/python/pacific_rim_protocol/formats/cdr.py
from __future__ import annotations

from ..encodings import WireEncoding, cdr_encoding


def cdr_format(type_name: str = "", *, package: str = "", schema_path: str = "") -> WireEncoding:
  """Compatibility wrapper for older callers.

  CDR is a wire encoding, not an IDL/data-format contract. Prefer
  `pacific_rim_protocol.cdr_encoding`.
  """
  return cdr_encoding(type_name, package=package, schema_path=schema_path)

File: infra/protocol/go/codec/formats/cdr.go
package formats

import protocolencodings "github.com/pacific-rim/pacific-rim/infra/protocol/go/codec/encodings"

// CDRFormat is a compatibility wrapper for older callers.
//
// Deprecated: CDR is a wire encoding, not an IDL/data-format contract. Prefer
// encodings.CDR and describe the source schema with ROS2MsgFormat,
// ROS2SrvFormat, ROS2IDLFormat, or ROS2TypeSupportFormat.
func CDRFormat(typeName string) DataFormat {
	return DataFormat{
		Kind:        DataFormatRawBytes,
		TypeName:    typeName,
		ContentType: protocolencodings.CDR(typeName).ResolvedContentType(),
		Metadata: map[string]string{
			"encoding": string(protocolencodings.WireEncodingCDR),
		},
	}
}
